jujugui/static/gui/src/app/store/env/controller-api.js

```diff
@@ -841,6 +841,138 @@ YUI.add('juju-controller-api', function(Y) {
         storageEndpoint: result['storage-endpoint'] || '',
         regions: regions
       };
+    },
+
+    /**
+      Returns the tags of cloud credentials for a set of users.
+
+      @method getTagsForCloudCredentials
+      @param {Array} userCloudPairs A list of (user-tag, cloud-tag) pairs for
+        which to retrieve the credentials, like
+        [['user-admin', 'cloud-google'], ['user-who@external', 'cloud-lxd']].
+      @param {Function} callback A callable that must be called once the
+        operation is performed. It will receive two parameters: an error (as a
+        string describing the problem) and a sequence of results. In the
+        sequence of results each result refers to the corresponding provided
+        user/cloud pair and holds an object with the following fields:
+        - err: a possible result specific error, in which case all subsequent
+          fields are omitted;
+        - tags: the list of tags that identify cloud credentials corresponding
+          to the user/cloud pair provided as input.
+        If no errors occur, error parameters are null. Otherwise, in case of
+        errors, the second argument is an empty array.
+    */
+    getTagsForCloudCredentials: function(userCloudPairs, callback) {
+      // Decorate the user supplied callback.
+      const handler = function(userCallback, data) {
+        if (!userCallback) {
+          console.log(
+            'data returned by Cloud.UserCredentials API call:', data);
+          return;
+        }
+        if (data.error) {
+          userCallback(data.error, []);
+          return;
+        }
+        const results = data.response.results;
+        if (!results) {
+          userCallback(null, []);
+          return;
+        }
+        const credentials = results.map(result => {
+          const err = result.error && result.error.message;
+          if (err) {
+            return {err: err};
+          }
+          return {tags: result.result || []};
+        });
+        userCallback(null, credentials);
+      }.bind(this, callback);
+      // Send the API request.
+      if (!userCloudPairs.length) {
+        userCloudPairs = [];
+      }
+      const userClouds = userCloudPairs.map(function(userCloud) {
```

Comment:
In case you weren't aware you could also do implicit returns with arrow fn's
```JavaScript
const userClouds = userCloudPairs.map(userCloud =>
  {'user-tag': userCloud[0], 'cloud-tag': userCloud[1]})
```
I have come to quite like them but I'll leave it up to you if you want to change it.